dotcom-rendering/src/web/components/Picture.tsx

```diff
@@ -1,320 +1,320 @@
-		// larger than in body showcase images so we use a different set of image sizes
-		if (breakpoint >= breakpoints.wide) return 1020;
-		if (breakpoint >= breakpoints.leftCol) return 940;
-		if (breakpoint >= breakpoints.tablet) return 700;
-		if (breakpoint >= breakpoints.phablet) return 700;
-		return breakpoint;
-	}
+const generateSignedUrl = ({
+	master,
+	imageWidth,
+	quality,
+}: {
+	master: string;
+	imageWidth: number;
+	quality: 'low' | 'high';
+}): string => {
+	const sign = (salt: string, path: string): string => {
+		return createHash('md5')
+			.update(salt + path)
+			.digest('hex');
+	};
 
-	if (isMainMedia && format.display === ArticleDisplay.Immersive) {
-		// Immersive main media images are often reported to have poor quality so we
-		// override the default values here and force larger images to be used
-		switch (breakpoint) {
-			case breakpoints.mobile:
-				return 480;
-			case breakpoints.mobileLandscape:
-				return 660;
-			case breakpoints.phablet:
-				return 740;
-			case breakpoints.tablet:
-				return 980;
-			case breakpoints.desktop:
-				return 1140;
-			case breakpoints.leftCol:
-				return 1300;
-			case breakpoints.wide:
-				return 1900;
-		}
+	const salt = process.env.IMAGE_SALT;
+	if (!salt && process.env.NODE_ENV === 'production' && !process.env.CI) {
+		// If no IMAGE_SALT env variable is set in production then something has
+		// gone wrong and we want to know about it
+		throw new Error(
+			'No IMAGE_SALT value found when constructing picture source',
+		);
+	} else if (!salt) {
+		// If no IMAGE_SALT is set outside of production it likely means the
+		// developer has yet to set the env file. In this case we continue
+		// using a default image
+		return 'https://i.guim.co.uk/img/media/52f8e9183bbc03c48fcdf4ed9a4e12b3a19a9927/0_0_5000_3000/master/5000.jpg?width=700&quality=45&auto=format&fit=max&dpr=2&s=196c7f25bf2fddcb45512d3ff25e20c0';
 	}
 
-	switch (role) {
-		case 'inline':
-			if (
-				breakpoint >= breakpoints.tablet &&
-				breakpoint < breakpoints.desktop
-			)
-				return 680;
-			if (breakpoint >= breakpoints.phablet) return 620;
-			return breakpoint;
-		case 'halfWidth':
-			if (breakpoint >= breakpoints.phablet) return 300;
-			return Math.round(breakpoint / 2);
-		case 'thumbnail':
-			return 140;
-		case 'immersive':
-			if (breakpoint >= breakpoints.wide) return 1300;
-			return breakpoint;
-		case 'supporting':
-			if (breakpoint >= breakpoints.wide) return 380;
-			if (breakpoint >= breakpoints.tablet) return 300;
-			return breakpoint;
-		case 'showcase':
-			if (breakpoint >= breakpoints.wide) return 860;
-			if (breakpoint >= breakpoints.leftCol) return 780;
-			if (breakpoint >= breakpoints.phablet) return 620;
-			return breakpoint;
-	}
+	// Construct and sign the url
+	const url = new URL(master);
+	const service = url.hostname.split('.')[0];
+	const params = new URLSearchParams({
+		width: imageWidth.toString(),
+		quality: quality === 'high' ? '45' : '85',
```

Comment:
The values `45` and `85` are tied to the VCL, so it would be good to hint towards where this information can be found. I can add the link if you reply with a 👍